### ISSUE TITLE
components.map is not a function

### ISSUE THREAD

**Issue by jhawlwut** [angular/universal-starter#87]
After cloning the repository and going through the steps, when I run "npm start", a handful of errors are generated.

```cmd
ERROR in /.../universal-starter/node_modules/@angular/router-deprecated/src/lifecycle/lifecycle_annotations.d.ts
    (29,100): error TS2304: Cannot find name 'Promise'.
```

This error is preceded by
```cmd
ts-loader: Using typescript@1.8.10 and /.../universal-starter/tsconfig.json
ts-loader: Using typescript@1.8.10 and /.../universal-starter/tsconfig.json
Hash: 6fc57360d385606425f14a98173e93e90b85b4f5
Version: webpack 1.13.0
Child
    Hash: 6fc57360d385606425f1
    Version: webpack 1.13.0
    Time: 8580ms
        Asset     Size  Chunks             Chunk Names
    bundle.js  2.25 MB       0  [emitted]  main
        + 382 hidden modules
```

Nodemon still runs the server, but when going to the address @ localhost:3000 the service crashes
```cmd
[nodemon] starting `node dist/server/bundle.js`
Listening on: http://localhost:3000
/.../universal-starter/node_modules/angular2-universal/dist/node/bootloader.js:246
        var directives = components.map(function (component) {
                                    ^

TypeError: components.map is not a function
    at Bootloader._applicationAll (/.../universal-starter/node_modules/angular2-universal/dist/node/bootloader.js:246:37)
    at Bootloader.serializeApplication (/.../universal-starter/node_modules/angular2-universal/dist/node/bootloader.js:78:21)
    at /.../universal-starter/node_modules/angular2-express-engine/dist/src/engine.js:71:38
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by gdi2290** [comment]
I updated universal to reflect better errors for this case. the problems seems to be that 1`directives: []` is empty. Please open another issue if this wasn't the case and you can reproduce this error again
